src/dodal/common/signal_utils.py

```diff
@@ -27,6 +31,40 @@ async def get_value(self) -> SignalDatatypeT:
         await self._update_value()
         return await super().get_value()
 
+    async def put(self, value: SignalDatatypeT | None, wait: bool) -> None:
+        if self.set_to_hardware_func:
+            write_value = self.initial_value if value is None else value
+            await self.set_to_hardware_func(write_value)
+        else:
+            return await super().put(value, wait)
+
+
+def create_rw_hardware_backed_soft_signal(
+    datatype: type[SignalDatatypeT],
+    get_from_hardware_func: Callable[[], Coroutine[Any, Any, SignalDatatypeT]],
+    set_to_hardware_func: SetHardwareType,
+    units: str | None = None,
+    precision: int | None = None,
+):
+    """Creates a soft signal that, when read will call the function passed into
+    `get_from_hardware_func` and return this. When set it will call `set_to_hardware_func`
+    and send something to the hardware.
+
+    This will allow you to make soft signals derived from arbitrary hardware signals.
+    However, calling subscribe on this signal does not give you a sensible value and
+    the signal is currently read only. See https://github.com/bluesky/ophyd-async/issues/525
+    for a more full solution.
+    """
+    return SignalRW(
+        backend=HardwareBackedSoftSignalBackend(
+            get_from_hardware_func,
+            set_to_hardware_func,
+            datatype,
+            units=units,
+            precision=precision,
+        )
+    )
+
 
 def create_hardware_backed_soft_signal(
```

Comment:
Could: Make it match soft_r /soft_rw ?
```suggestion
def create_r_hardware_backed_soft_signal(
```